docs/components/concepts/execution-listeners.md

```diff
@@ -0,0 +1,352 @@
+    end
+
+    ElementActivation -->|if 'start' ELs configured| StartExecutionListeners
+    StartExecutionListeners:::execution-listeners --> FinalizeActivation
+    ElementActivation --> FinalizeActivation
+    FinalizeActivation --> ElementExecution
+    ElementExecution --> ElementCompletion
+    ElementCompletion -->|if 'end' ELs configured| EndExecutionListeners
+    EndExecutionListeners:::execution-listeners --> FinalizeCompletion
+    ElementCompletion --> FinalizeCompletion
+
+    classDef execution-listeners fill:gray
+```
+
+<details>
+  <summary><strong>Element lifecycle steps</strong></summary>
+  <p>When an element with ELs is processed in Zeebe, the following steps outline its lifecycle, highlighting where <code>start</code> and <code>end</code> ELs are executed:</p>
+  <h4>Element Activation:</h4>
+  <p>The process instance reaches the element and begins its activation. This initial phase prepares the element for execution and typically includes:</p>
+  <ul>
+    <li><em>Apply Input Mappings</em> (if applicable): Input variables are mapped to the element's local scope.</li>
+  </ul>
+  <h4>Start Execution Listeners:</h4>
+  <p>If <code>start</code> ELs are defined, they are executed. These listeners can perform setup tasks and have access to input variables (if any defined).</p>
+  <h4>Finalize Activation:</h4>
+  <p>The element completes any additional actions required for activation. This usually includes:</p>
+  <ul>
+    <li><em>Subscribe to Events</em> (if applicable): The element subscribes to any events it needs to listen for during its execution.</li>
+    <li><em>Element Activated</em>: The element is fully activated and ready to perform its primary function.</li>
+  </ul>
+  <h4>Element Execution:</h4>
+  <p>The core activity of the element is carried out. This step represents the main purpose of the BPMN element (e.g., a task being performed).</p>
+  <h4>Element Completion:</h4>
+  <p>Once the primary activity is finished, the element begins its completion phase, which includes:</p>
+  <ul>
+    <li><em>Apply Output Mappings</em> (if applicable): Output variables are mapped from the element's local scope to the global scope.</li>
+    <li><em>Unsubscribe from Events</em> (if applicable): The element unsubscribes from any events it was listening for.</li>
+  </ul>
+  <h4>End Execution Listeners:</h4>
+  <p>During the element completion stage, if <code>end</code> ELs are defined, they are executed after applying output mappings (if any defined). These listeners can perform cleanup tasks and have access to output variables.</p>
+  <h4>Finalize Completion:</h4>
+  <p>The element executes any additional actions required for element completion and execution transits to the next element.</p>
+  <p>By placing custom logic at strategic points within the element lifecycle, you can enhance process control, improve data handling, and integrate seamlessly with external systems.</p>
+</details>
```

Comment:
:wrench: This part is relevant to use EL correctly. We should not hide it.  

:wrench: We should make this part less technical and easier to understand for users.